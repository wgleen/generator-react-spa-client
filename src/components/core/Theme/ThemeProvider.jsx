import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider as ThemeProviderStyled } from 'styled-components'

export const theme = {
  typography: {
    fontFamilyPrimary: 'Open Sans',
    fontWeightPrimary: 300,
    lineHeight: 'normal',
    textAlign: 'left'
  },
  colors: {
    primaryDarkest: '#860000',
    primaryDark: '#b90000',
    primary: '#ec0000',
    primaryLight: '#ff8686',
    primaryLightest: '#ffd2d2',
    secondaryDarkest: '#247a86',
    secondaryDark: '#2e9faf',
    secondary: '#43bccd',
    secondaryLight: '#94d9e2',
    secondaryLightest: '#d0eef2',
    grayDarkest: '#0d0d0d',
    grayDark: '#4d4d4d',
    gray: '#737373',
    grayLight: '#b3b3b3',
    grayLightest: '#e5e5e5',
    red: '#ff3225',
    yellow: '#e6e328',
    green: '#3cab3a',
    blue: '#2e7baf'
  }
}

const ThemeProvider = (props) => {
  const { children } = props

  return <ThemeProviderStyled theme={theme}>{children}</ThemeProviderStyled>
}

ThemeProvider.defaultProps = {
  children: null
}

ThemeProvider.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node])
}

export default ThemeProvider
