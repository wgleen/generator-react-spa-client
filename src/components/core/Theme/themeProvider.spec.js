import React from 'react'
import { mount } from 'enzyme'
import ThemeProvider from './ThemeProvider'

describe('<ThemeProvider />', () => {
  it('should render correctly', () => {
    const component = mount(<ThemeProvider />)

    expect(component).toMatchSnapshot()

    component.unmount()
  })
})
