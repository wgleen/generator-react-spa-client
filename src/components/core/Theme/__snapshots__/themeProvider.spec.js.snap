// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeProvider /> should render correctly 1`] = `
<ThemeProvider>
  <Component
    theme={
      Object {
        "colors": Object {
          "blue": "#2e7baf",
          "gray": "#737373",
          "grayDark": "#4d4d4d",
          "grayDarkest": "#0d0d0d",
          "grayLight": "#b3b3b3",
          "grayLightest": "#e5e5e5",
          "green": "#3cab3a",
          "primary": "#ec0000",
          "primaryDark": "#b90000",
          "primaryDarkest": "#860000",
          "primaryLight": "#ff8686",
          "primaryLightest": "#ffd2d2",
          "red": "#ff3225",
          "secondary": "#43bccd",
          "secondaryDark": "#2e9faf",
          "secondaryDarkest": "#247a86",
          "secondaryLight": "#94d9e2",
          "secondaryLightest": "#d0eef2",
          "yellow": "#e6e328",
        },
        "typography": Object {
          "fontFamilyPrimary": "Open Sans",
          "fontWeightPrimary": 300,
          "lineHeight": "normal",
          "textAlign": "left",
        },
      }
    }
  />
</ThemeProvider>
`;
